diff -up musl-1.2.4/configure.1~ musl-1.2.4/configure
--- musl-1.2.4/configure.1~	2023-05-02 16:37:02.162912602 +0200
+++ musl-1.2.4/configure	2023-05-02 16:37:59.011416485 +0200
@@ -361,6 +361,28 @@ tryflag CFLAGS_C99FSE -frounding-math
 #
 tryflag CFLAGS_C99FSE -fno-strict-aliasing
 
+if [ "$malloc_dir" = "mimalloc" ]; then
+	# mimalloc needs access to headers that are skipped by
+	# -nostdinc. We still don't want to see system libc headers,
+	# but we want to see compiler headers such as stdatomic.h
+	# and emmintrin.h
+	#
+	# Unfortunately there is no standard way to get the compiler
+	# header search path, but this works for (at least) clang
+	# and gcc:
+	printf "checking for compiler intrinsic headers path... "
+	INCPATH="$($CC -print-search-dirs |grep ^libraries: |cut -d= -f2 |cut -d: -f1)/include"
+	if ! [ -e "$INCPATH/stdatomic.h" ]; then
+		printf "not found\n"
+		printf "Couldn't detect stdatomic.h for your compiler. Either add\n"
+		printf "a detection method or use a malloc implementation other than\n"
+		printf "mimalloc.\n"
+		exit 1
+	fi
+	printf "$INCPATH\n"
+	tryflag CFLAGS_C99FSE "-isystem $INCPATH"
+fi
+
 #
 # We may use the may_alias attribute if __GNUC__ is defined, so
 # if the compiler defines __GNUC__ but does not provide it,
diff -up musl-1.2.4/src/malloc/mimalloc/glue.h.1~ musl-1.2.4/src/malloc/mimalloc/glue.h
--- musl-1.2.4/src/malloc/mimalloc/glue.h.1~	2023-05-02 16:37:02.162912602 +0200
+++ musl-1.2.4/src/malloc/mimalloc/glue.h	2023-05-02 16:37:02.162912602 +0200
@@ -0,0 +1,15 @@
+#ifndef MALLOC_GLUE_H
+#define MALLOC_GLUE_H
+
+#define mi_malloc __libc_malloc_impl
+#define mi_realloc __libc_realloc
+#define mi_free __libc_free
+
+//#define malloc __libc_malloc_impl
+//#define realloc __libc_realloc
+//#define free __libc_free
+
+#define mmap __mmap
+#define madvise __madvise
+#define mremap __mremap
+#endif
